app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -65,25 +67,24 @@ class SyncIndicatorViewController @Inject constructor(
     return showSyncIndicatorState.mergeWith(showDefaultSyncIndicatorState)
   }
 
-  private fun syncedFailureState(timestamp: Instant?): SyncIndicatorState {
+  private fun syncedFailureState(timestamp: Instant?, maxIntervalSinceLastSync: Duration): SyncIndicatorState {
     if (timestamp == null) {
       return SyncPending
     }
 
-    return syncIndicatorState(timestamp)
+    return syncIndicatorState(timestamp, maxIntervalSinceLastSync)
   }
 
-  private fun syncIndicatorState(timestamp: Instant?): SyncIndicatorState {
+  private fun syncIndicatorState(timestamp: Instant, syncFailureThreshold: Duration): SyncIndicatorState {
```

Comment:
It feels strange that we have two functions here that only differ in nullability of one param. 

@vinaysshenoy can you take a look at this? Does this look strange to you as well?

Reply:
I wanted to separate the nullability check because this will happen only in case of failure.
I could move the check to inside the `when` block itself.
I just felt this was more readable.

Reply:
You're right, it does seem a little weird. Maybe we should consider merging the two methods.

Reply:
> I wanted to separate the nullability check because this will happen only in case of failure.

I see. In this specific case, it feels like merging sounds better as @vinaysshenoy said

Reply:
Done